lms/static/scripts/frontend_apps/components/EmailDigestPreferences.tsx

```diff
@@ -0,0 +1,101 @@
+import { Button, Checkbox } from '@hypothesis/frontend-shared';
+import classnames from 'classnames';
+import { useCallback } from 'preact/hooks';
+
+import type { SelectedDays, WeekDay } from '../config';
+
+const dayNames: [WeekDay, string][] = [
+  ['sun', 'Sunday'],
+  ['mon', 'Monday'],
+  ['tue', 'Tuesday'],
+  ['wed', 'Wednesday'],
+  ['thu', 'Thursday'],
+  ['fri', 'Friday'],
+  ['sat', 'Saturday'],
+];
+
+export type EmailDigestPreferencesProps = {
+  /** Currently selected days */
+  selectedDays: SelectedDays;
+  /** Callback to fully or partially update currently selected days, without saving */
+  onSelectedDaysChange: (newSelectedDays: Partial<SelectedDays>) => void;
+};
+
+export default function EmailDigestPreferences({
+  selectedDays,
+  onSelectedDaysChange,
+}: EmailDigestPreferencesProps) {
+  const setAllTo = useCallback(
+    (enabled: boolean) =>
+      onSelectedDaysChange({
+        sun: enabled,
+        mon: enabled,
+        tue: enabled,
+        wed: enabled,
+        thu: enabled,
+        fri: enabled,
+        sat: enabled,
+      }),
+    [onSelectedDaysChange],
+  );
+  const selectAll = useCallback(() => setAllTo(true), [setAllTo]);
+  const selectNone = useCallback(() => setAllTo(false), [setAllTo]);
+
+  return (
+    <div className="flex flex-col gap-y-3">
+      <h2 className="text-lg/4 text-brand">Student activity digest</h2>
```

Comment:
There is some duplication of the styling for form components across `EmailDigestPreferences` and `EmailMentionsPreferences`. I think it would be worth extracting some common components, probably separate from this PR, so we can define the styles in one place.